src/extensions/extension-store.ts

```diff
@@ -26,13 +26,13 @@ import type { LensExtension } from "./lens-extension";
 export abstract class ExtensionStore<T> extends BaseStore<T> {
   protected extension: LensExtension;
 
-  async loadExtension(extension: LensExtension) {
+  loadExtension(extension: LensExtension) {
```

Comment:
Though this is technically a breaking change (which I can easily revert) I think that it is better to leave it as synchronous so that the extension developers can gain the same benefits as ourselves.